src/geohash_helper.c

```diff
@@ -233,8 +231,10 @@ double geohashGetDistance(double lon1d, double lat1d, double lon2d, double lat2d
     lon1r = deg_rad(lon1d);
     lon2r = deg_rad(lon2d);
     v = sin((lon2r - lon1r) / 2);
-    /* if v == 0 we can avoid doing expensive math when lons are practically the same */
-    if (v == 0.0) return geohashGetLatDistance(lat1d, lat2d);
+    /* Reflects about 6nm on earth for comparing longitudes. */
+    const double GEO_EPSILON = 1e-15;
+    /* if v == 0, or practically 0, we can avoid doing expensive math when lons are practically the same */
```

Comment:
This is not only to avoid expensive math. It is also to compensate for rounding errors. Could be mentioned in this comment.

Reply:
I originally had a comment, but comparing double to zero is already not a best practice because of rounding errors. I try to avoid comments I think are supposed to be obvious.